cloudinit/distros/alpine.py

```diff
@@ -202,6 +205,344 @@ def preferred_ntp_clients(self):
+
+        # Find the line in /etc/shadow for the user
+        original_line = None
+        for line in shadow_contents.splitlines():
+            current_user = line.split(":")[0]
+            if current_user == name:
+                original_line = line
+
+        if original_line is not None:
+            # Modify field(s) in copy of user's shadow file entry
+            new_list = original_line.split(":")
+            update_type = ""
+
+            # Minimum password age
+            new_list[3] = ""
+            # Maximum password age
+            new_list[4] = ""
+            # Password warning period
+            new_list[5] = ""
+            update_type = "password aging"
+
+            if expiredate is not None:
+                # Convert date into number of days since 1st Jan 1970
+                days = (
+                    datetime.fromisoformat(expiredate)
+                    - datetime.fromisoformat("1970-01-01")
+                ).days
+                new_list[7] = str(days)
+                if update_type != "":
+                    update_type = update_type + " & "
+                update_type = update_type + "acct expiration date"
+            if inactive is not None:
+                new_list[6] = inactive
+                if update_type != "":
+                    update_type = update_type + " & "
+                update_type = update_type + "inactivity period"
+
+            new_line = ":".join(new_list)
+
+            # Replace existing line for user with modified line
+            shadow_contents = shadow_contents.replace(original_line, new_line)
+            LOG.debug("Updating %s for user %s", update_type, name)
+            try:
+                util.write_file(
+                    shadow_file, shadow_contents, omode="w", preserve_mode=True
+                )
+            except IOError as e:
+                util.logexc(LOG, "Failed to update %s file", shadow_file)
+                raise e
+        else:
+            util.logexc(
+                LOG, "Failed to update %s for user %s", shadow_file, name
+            )
+
+    def lock_passwd(self, name):
+        """
+        Lock the password of a user, i.e., disable password logins
+        """
+
+        # Check whether Shadow's or Busybox's version of 'passwd'.
+        # If Shadow's 'passwd' is available then use the generic
+        # lock_passwd function from __init__.py instead.
+        (_out, err) = subp.subp(["passwd", "--help"])
+        if not err.startswith("BusyBox"):
+            return super().lock_passwd(name)
+
+        cmd = ["passwd", "-l", name]
+        # Busybox's 'passwd', unlike Shadow's 'passwd', errors
+        # if password is already locked:
+        #
+        #   "passwd: password for user2 is already locked"
+        #
+        # with exit code 1
+        try:
+            (_out, err) = subp.subp(cmd, rcs=[0, 1])
+            if re.search(r"is already locked", err):
+                return True
+        except Exception as e:
```

Comment:
```suggestion
        except subp.ProcessExecutionError as e:
```

Reply:
Done.

Similar issue exists in lock_passwd in __init__.py so, like some of @TheRealFalcon previous comments, the same improvements could be made there also